### ISSUE TITLE
[Bug] Erroneous "Account does not exist" displays when an account has no SOL

### ISSUE THREAD

**Issue by JuanMarchetto** [solana-labs/explorer#336]
**Describe the bug**
 "Account does not exist" is get displays when an account has no SOL, even if is an account with previus activity and has ATAs

**To Reproduce**

1. Look up an address with some SOL and at least another token or NFT for more clarity
2. Check how the balance is displayed on the explorer
3. Send max amount of SOL to another wallet you own
4. You can verify that now how the explorer displays "Account does not exist" in the "Balance (SOL)" field

**Example Links**

-   [explorer](https://explorer.solana.com)

**Expected behavior**
An account should be display his SOL balance as 0 in this escenario

**Screenshots**
<img width="1433" alt="Screenshot 2024-04-20 at 09 47 48" src="https://github.com/solana-labs/explorer/assets/6706152/6cd48038-4bb9-41e4-88fe-0c7a9b29927f">

**Comment by JuanMarchetto** [comment]
I can make a PR to address this issue if you agree

**Comment by nickfrosty** [comment]
This is as expected. When an account did exist and all its SOL (aka lamports) are transferred out, making its balance 0, the account gets garbage collected by the run time. Making it no longer exist

**Comment by JuanMarchetto** [comment]
> This is as expected. When an account did exist and all its SOL (aka lamports) are transferred out, making its balance 0, the account gets garbage collected by the run time. Making it no longer exist

Ok, i understand, maybe is a too technical concept for general users do, but if you think that display that is enough we can close this issue

**Comment by philetos** [comment]
@nickfrosty what about for cNFT Accounts?  https://github.com/solana-labs/explorer/issues/378

**Comment by J4Web** [comment]
@JuanMarchetto  Technically, that’s correct. However, it could be confusing for users who recently swapped all of their SOL (lamports) to another token like JUP. When they check their account on the explorer and see ‘Account not found,’ it might feel alarming. A clearer message could help prevent this confusion what do you think ?

**Comment by nickfrosty** [comment]
@philetos cnfts are not technically in accounts though. The way their hash id works looks similar because it's base58 encoded, but it functions very different.

@J4Web it could be alarming/concerning. A more clear message might be warranted though

**Comment by rogaldh** [comment]
I agree that the wording is a bit technical, but that is how it is. In the moment, when there is no SOL, account does not exist. Tinkering with the phrasing won't improve that, I think, and might confuse on the other hand.

@Woody4618 I think this could be closed unless you have better idea. WDYT?

**Comment by JuanMarchetto** [comment]
I would argue that if what is being shown is the balance in sol, as the key indicates, then the correct value to display is zero and the text currently used is an additional technical observation and not the concrete value of that key.

**Comment by Woody4618** [comment]
We can change the text to: 
"This address currently has no system account on-chain because it holds 0 lamports. It may still have associated token accounts, NFTs, or previous activity."

**Comment by JuanMarchetto** [comment]
> We can change the text to: "This address currently has no system account on-chain because it holds 0 lamports. It may still have associated token accounts, NFTs, or previous activity."

I think this is a good solution

**Comment by Woody4618** [comment]
Can you make a pr?

**Comment by JuanMarchetto** [comment]
Yes, I'll do it
